Troubleshooting: Update statement-summary-tables.md (#2777)
pingcap/docs: statement-summary-tables.md

- `tidb_stmt_summary_refresh_interval`: The interval at which the `statements_summary` table is refreshed. The time unit is second (s). The default value is `1800`.
- `tidb_stmt_summary_history_size`: The size of each SQL statement category stored in the `statements_summary_history` table. The default value is `24`.
- `tidb_stmt_summary_max_stmt_count`: Limits the number of SQL statements that can be stored in statement summary tables. The default value is `200`. If the limit is exceeded, those SQL statements that recently remain unused are cleared.
- `tidb_stmt_summary_max_sql_length`: Specifies the longest display length of `DIGEST_TEXT` and `QUERY_SAMPLE_TEXT`. The default value is `4096`.
- `tidb_stmt_summary_internal_query`: Determines whether to count the TiDB SQL statements. `1` means `count`, 0 means `not count`. The default value is `0`.

> The following shows a configuration example:

Reviewer: ```suggestion
An example of the statement summary configuration is shown as follows:
```

{{< copyable "sql" >}}

```sql
set global tidb_enable_stmt_summary = true;
set global tidb_stmt_summary_refresh_interval = 1800;
set global tidb_stmt_summary_history_size = 24;
```

> When the above configuration takes effect, every 30 minutes the `statements_summary` is cleared. `statements_summary_history` stores data generated over the recent 12 hours.

Reviewer: ```suggestion
After the configuration above takes effect, every 30 minutes the `statements_summary` table is cleared. The `statements_summary_history` table stores data generated over the recent 12 hours.
```

> The above system variables have two scopes: global and session. These scopes work differently from other system variables:

Reviewer: ```suggestion
The system variables above have two scopes: global and session. These scopes work differently from other system variables:
```

> - Set the global variable to apply to the cluster immediately.

Reviewer: ```suggestion
- After setting the global variable, your setting applies to the whole cluster immediately.
```

> - Set the session variable to apply to the current TiDB server immediately. This is useful when you debug on a single TiDB server instance.

Reviewer: ```suggestion
- After setting the session variable, your setting applies to the current TiDB server immediately. This is useful when you debug on a single TiDB server instance.
```

- The session variable has a higher read priority. The global variable is read only when no session variable is set.

> - Set the session variable to a blank string to re-read the global variable.

Reviewer: ```suggestion
- If you set the session variable to a blank string, the global variable is re-read.
```

> **Note:**
>
> The `tidb_stmt_summary_history_size`, `tidb_stmt_summary_max_stmt_count`, and `tidb_stmt_summary_max_sql_length` configuration items affect memory usage. It is recommended that you adjust these configurations based on your needs. It is not recommended to set them too large values.

> ## Limitations

Reviewer: ```suggestion
## Limitation
```

> The statement summary tables have the following limitations:

Reviewer: ```suggestion
The statement summary tables have the following limitation:
```

> The statement summary data are lost when the TiDB server is restarted. This is because statement summary tables are memory tables, and the data is cached in memory instead of being persisted on storage.

Reviewer: ```suggestion
The statement summary data will be lost when the TiDB server is restarted. This is because statement summary tables are memory tables, and the data is cached in memory instead of being persisted on storage.
```

## Troubleshooting examples

> This section shows how to use the statement summary feature to troubleshoot SQL performance issues.

Reviewer: ```suggestion
This section provides two examples to show how to use the statement summary feature to troubleshoot SQL performance issues.
```

### Could high SQL latency be caused by the server end?

> In this example, the client shows slow performance with point queries on the employee table. You can perform a fuzzy search on SQL texts:

Reviewer: ```suggestion
In this example, the client shows slow performance with point queries on the `employee` table. You can perform a fuzzy search on SQL texts:
```

{{< copyable "sql" >}}

```sql
SELECT avg_latency, exec_count, query_sample_text
    FROM information_schema.statements_summary
    WHERE digest_text LIKE 'select * from employee%';
```

 `1ms` and `0.3ms` are considered within the normal range of `avg_latency`. Therefore, it can be concluded that the server end is not the cause. You can troubleshoot with the client or the network.

{{< copyable "sql" >}}

> ```

Reviewer: ```suggestion
```sql
```

+-------------+------------+------------------------------------------+
| avg_latency | exec_count | query_sample_text                        |
+-------------+------------+------------------------------------------+
|     1042040 |          2 | select * from employee where name='eric' |
|      345053 |          3 | select * from employee where id=3100     |
+-------------+------------+------------------------------------------+
2 rows in set (0.00 sec)
```

### Which categories of SQL statements consume the longest total time?

> If the QPS decreases significantly from 10:00 A.M. to 10:30 A.M., you can find out the three categories of SQL statements with the longest time consumption from the history table:

Reviewer: QPS means `queries per second`.
```suggestion
If the QPS decrease significantly from 10:00 to 10:30, you can find out the three categories of SQL statements with the longest time consumption from the history table:
```

{{< copyable "sql" >}}

```sql
SELECT sum_latency, avg_latency, exec_count, query_sample_text
    FROM information_schema.statements_summary_history
    WHERE summary_begin_time='2020-01-02 10:00:00'
    ORDER BY sum_latency DESC LIMIT 3;
```

The result shows that the following three categories of SQL statements consume the longest time in total, which need to be optimized with high priority.

{{< copyable "sql" >}}

> ```

Reviewer: ```suggestion
```sql
```

- `SAMPLE_USER`: The users who execute SQL statements of this category. Only one user is taken.
- `PLAN_DIGEST`: The digest of the execution plan.
- `PLAN`: The original execution plan. If there are multiple statements, the plan of only one statement is taken.
- `PLAN_CACHE_HITS`: The total number of times that SQL statements of this category hit the plan cache.

> - `PLAN_IN_CACHE`: Determines whether the previous execution of SQL statements of this category hit the plan cache.

Reviewer 2: Would `indicates` be better? `determines` may be regarded as `decides`.

Author: updated

> - `PLAN_IN_CACHE`: Determines whether the previous execution of SQL statements of this category hit the plan cache.

Reviewer: ```suggestion
- `PLAN_IN_CACHE`: Indicates whether the previous execution of SQL statements of this category hit the plan cache.
```
